extutils/imgproc/apng2png.py

```diff
@@ -0,0 +1,285 @@
+
+    length: int = field(init=False)
+    data: bytes = field(init=False)
+    crc: str = field(init=False)
+
+    def __post_init__(self):
+        self.length = self.base_chunk.length
+        self.data = self.base_chunk.data
+        self.crc = self.base_chunk.crc
+
+    @property
+    def pure_data(self) -> bytes:
+        """
+        Get the data excluding length, type and CRC.
+
+        :return: data excluding length, type and CRC
+        """
+        return self.base_chunk.pure_data
+
+
+@dataclass
+class _DataChunkIHDR(_DataChunkBase):
+    width: int = field(init=False)
+    height: int = field(init=False)
+
+    def __post_init__(self):
+        super().__post_init__()
+
+        self.width, self.height = struct.unpack(">II", self.pure_data[0:8])
+
+    def replace_resolution(self, new_width: int, new_height: int) -> '_DataChunkIHDR':
+        new_data = struct.pack(">II", new_width, new_height) + self.pure_data[8:]
+
+        return _DataChunkIHDR(_BaseChunk.make_chunk("IHDR", new_data))
+
+
+@dataclass
+class _DataChunkacTL(_DataChunkBase):
+    frame_count: int = field(init=False)
+
+    def __post_init__(self):
+        super().__post_init__()
+
+        self.frame_count = struct.unpack(">I", self.pure_data[0:4])[0]
+
+
+@dataclass
+class _DataChunkIDAT(_DataChunkBase):
+    pass
+
+
+@dataclass
+class _DataChunkIEND(_DataChunkBase):
+    pass
+
+
+@dataclass
+class _DataChunkOther(_DataChunkBase):
+    pass
+
+
+@dataclass
+class _DataChunkfcTL(_DataChunkBase):
+    seq_num: int = field(init=False)
+    width: int = field(init=False)
+    height: int = field(init=False)
+    x_offset: int = field(init=False)
+    y_offset: int = field(init=False)
+    delay_num: int = field(init=False)
+    delay_den: int = field(init=False)
+    dispose: _Dispose = field(init=False)
+    blend: int = field(init=False)
+
+    def __post_init__(self):
+        super().__post_init__()
+
+        self.seq_num, self.width, self.height, self.x_offset, self.y_offset = \
+            struct.unpack(">IIIII", self.pure_data[0:20])
+        self.delay_num, self.delay_den, self.dispose, self.blend = \
+            struct.unpack(">HHBB", self.pure_data[20:26])
+
+        self.dispose = _Dispose(self.dispose)
+
+    # noinspection PyPep8Naming
+    def generate_IHDR(self, IHDR: _DataChunkIHDR) -> _DataChunkIHDR:
+        return IHDR.replace_resolution(self.width, self.height)
+
+
+@dataclass
+class _DataChunkfdAT(_DataChunkBase):
+    # noinspection PyPep8Naming
+    def to_IDAT(self):
+        # Skipping 4 B because it's frame number
+        return _DataChunkIDAT(_BaseChunk.make_chunk("IDAT", self.base_chunk.pure_data[4:]))
+
+
+def _parse_chunks(data) -> Generator[_BaseChunk, None, None]:
+    data = data[8:]
+
+    while data:
+        new_chunk = _BaseChunk.from_bytes(data)
+        yield new_chunk
+
+        data = data[new_chunk.length + 12:]
+
+
+# noinspection PyPep8Naming
+def _create_image_bytes(IHDR: _DataChunkIHDR, fcTL: _DataChunkfcTL,
```

Comment:
Codacy found an issue: [Argument name "IHDR" doesn't conform to snake_case naming style](https://app.codacy.com/manual/RaenonX/Jelly-Bot/pullRequest?prid=6015490)